WordPress/Classes/ViewRelated/Blog/SharingViewController.m

```diff
@@ -101,32 +133,65 @@ - (void)doneButtonTapped
     [self dismissViewControllerAnimated:YES completion:nil];
 }
 
+#pragma mark - Table view sections
+
+- (void)refreshSections {
+    NSMutableArray<NSNumber *> *sections = [NSMutableArray new];
+
+    if ([self.supportedServices count] > 0) {
+        [sections addObject:@(SharingSectionAvailableServices)];
+    }
+
+    if (self.unsupportedServices.count > 0) {
+        [sections addObject:@(SharingSectionUnsupported)];
+    }
+
+    if ([self.blog supportsShareButtons]) {
+        [sections addObject:@(SharingSectionSharingButtons)];
+    }
+
+    self.sections = sections;
+}
+
+- (SharingSectionType)sectionTypeForIndex:(NSInteger)index
+{
+    if (index >= self.sections.count) {
+        return SharingSectionUndefined;
+    }
+
+    return [self.sections objectAtIndex:index].intValue;
+}
+
 #pragma mark - UITableView Delegate methods
 
 - (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
 {
-    NSInteger count = SharingSectionCount;
-    if (![self.blog supportsShareButtons]) {
-        count -= 1;
-    }
-    return count;
+    return self.sections.count;
 }
 
 - (NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section
 {
-    switch (section) {
-        case SharingPublicizeServices:
+    SharingSectionType sectionType = [self sectionTypeForIndex:section];
+    switch (sectionType) {
+        case SharingSectionAvailableServices:
             return NSLocalizedString(@"Jetpack Social Connections", @"Section title for Publicize services in Sharing screen");
-        case SharingButtons:
+
+        case SharingSectionUnsupported:
+            return NSLocalizedString(@"Twitter Auto-sharing Is No Longer Available",
+                                     @"Section title for the disabled Twitter service in the Social screen");
```

Comment:
Good catch. I intended to update this to a key format but forgot to do it since it uses a different method name. Anyways, this is now addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/20708/commits/21c68f8d5d3334a501860aee531bf9a9da36986c. Thanks Chris!